ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -1623,6 +1623,16 @@ static size_t tsrm_tls_offset;
 ||	}
 |.endmacro
 
+|.macro ENDBR
+||#if defined (__CET__) && (__CET__ & 1) != 0
+|	.if X64
+|		endbr64
+|	.else
+|		endbr32
+|	.endif
+||#endif
+|.endmacro
+
```

Comment:
Can we enable/disable generation of ``endbr`` ar run-time? e.g. thorough new bit in ``opcache.jit`` directive?

Reply:
I'm not sure if this is really necessary... 
Can you share some CET vs non-CET benchmarks results? (at least Zend/bench.php)

Reply:
My concern is that runtime enable/disable may cause inconsistency, i.e., jitted code has `endbr` and gcced code doesn't. Our current method mimics or aligns with the behavior of gcc who enable/disable generation of `endbr` at compile time.
**GCC:**
When compiled with `gcc -fcf-protection=full` (default) or `gcc -fcf-protection=branch`, the MACRO `__CET__` is defined and `endbr` is inserted. It will also emit a property named `IBT` to ELF header.
**Execution flow:**
Once ld.so see `IBT` property in ELF header, it issues an arch_prctl() syscall to kernel to enable HW/CPU CET engine.
**Benchmarks results:**
I will run and share Google PKB benchmark result next week. Below are quick Zend/bench.php results:

- CET. Binary has endbr and HW/CPU CET on.

```
$ ./sapi/cli/php Zend/bench.php
simple             0.026
simplecall         0.011
simpleucall        0.031
simpleudcall       0.026
mandel             0.098
mandel2            0.087
ackermann(7)       0.018
ary(50000)         0.003
ary2(50000)        0.002
ary3(2000)         0.033
fibo(30)           0.056
hash1(50000)       0.006
hash2(500)         0.007
heapsort(20000)    0.022
matrix(20)         0.021
nestedloop(12)     0.022
sieve(30)          0.012
strcat(200000)     0.004
------------------------
Total              0.485
```

- Non-CET.  Binary has no endbr and HW/CPU CET off

```
$ ./sapi/cli/php Zend/bench.php
simple             0.025
simplecall         0.011
simpleucall        0.030
simpleudcall       0.025
mandel             0.098
mandel2            0.090
ackermann(7)       0.020
ary(50000)         0.003
ary2(50000)        0.003
ary3(2000)         0.031
fibo(30)           0.056
hash1(50000)       0.006
hash2(500)         0.007
heapsort(20000)    0.022
matrix(20)         0.021
nestedloop(12)     0.022
sieve(30)          0.012
strcat(200000)     0.004
------------------------
Total              0.486
```

BTW, when HW/CPU CET engine is OFF, the `endbr` is regarded as `nop`. For this case, even the app has `endbr` inside, the perf impact is negligible.